python/whylogs/core/schema.py

```diff
@@ -76,6 +85,7 @@ def __init__(
         schema_based_automerge: bool = False,
         segments: Optional[Dict[str, SegmentationPartition]] = None,
         validators: Optional[Dict[str, List[Validator]]] = None,
+        augmenters: List[AugmenterSpec] = [],
```

Comment:
Default value here should not be an empty list (current implementation will create a shared instance across instances of DatasetSchema). Add default_factory here to get separate instances of the list for each DatasetSchema instance.

You may need to add imports for `field` and `default_factory`
```suggestion
        augmenters: List[AugmenterSpec] = field(default_factory=list),
```